### PR TITLE
When cloning elements, clone events from all child elements

### PR DESCRIPTION
**Pull request by benvinegar**
Somehow #65 became closed – here's part 2.

Per @rvagg's suggestion, I converted integration.html into a proper sink test suite. Then I added a test case that demonstrates this bug.

### CODE REVIEW

**Comment by rvagg** [comment]
Nice work on the integration tests! We'll have to start coming up with more tests for the cross-over points within the Jeesh, such as the Qwery->Bonzo dependency for `$('<>')` element creation.

FYI, the font on the test page goes default serif on systems without Helvetica Neue, such as mine, not that this is a problem but it could be reset at the end of that test with `$('body').css('font', '')`.

**Comment by fat** [comment]
trying to run the integration tests and i get an error : `String.prototype.toString is not generic` any ideas?

apparently you can't $(el).addClass(0)...

File: integration/integration.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+sink('Ender', function(test, ok, before, after) {
+  function byid(id) { return document.getElementById(id); }
+
+  test('classes', 2, function() {
+    var parents = $('#get-me').parents('div.robustr').addClass('aloha');
+    parents.each(function (el, i) {
+      $(el).addClass(i);
```
Comment (fat) [inline integration/integration.js]: doesn't work for me... have to change this to addClass(i + ''). Or we should change bonzo to handle passing 0 in. Or i'm a n00b and missing somethign obvious.

File: bonzo.js [hunk 2]
```diff
@@ -171,10 +171,22 @@
...
             var c = el.cloneNode(true)
+              , cloneElems
+              , elElems;
+
             // check for existence of an event cloner
             // preferably https://github.com/fat/bean
             // otherwise Bonzo won't do this for you
-            self.$ && self.cloneEvents && self.$(c).cloneEvents(el)
+            if (self.$ && self.cloneEvents) {
+              self.$(c).cloneEvents(el);
```
Comment (fat) [inline bonzo.js]: probably should remove semicolons to be consistent...

**Comment by fat** [comment]
fix that stuff up and everything looks good to me!

**Comment by benvinegar** [comment]
Made those changes.

**Comment by rvagg** [comment]
did you commit? still seeing only your commits from last month.

**Comment by benvinegar** [comment]
I amended the commits.

**Comment by rvagg** [comment]
oh, that's interesting, I thought pushing amended commits to GitHub didn't work, at least they always have for me, I'll have to play with that.

**Comment by rvagg** [comment]
.. and, I've just noticed that you've modified bonzo.js rather than src/bonzo.js, you may want to change that or your changes will disappear after a make.

**Comment by benvinegar** [comment]
Fixed. Also, looks like my test wasn't covering the right scenario 100%. 

Aside: you're right that GH pull requests don't really care for amending – won't do that again.

**Comment by ded** [comment]
this wasn't the first PR that changed `bonzo.js` and not `src/bonzo.js` — i really need to document this better in the readme

**Comment by fat** [comment]
This all looks good - @ded any reason not to merge?

**Comment by ded** [comment]
i'll pull and run the tests.

**Comment by ded** [comment]
lol. i just tried `git pull https://github.com/benvinegar/bonzo clone-events` — and it couldn't find the repo because you named it `funzo`.... right on. trying again.

**Pull request merged**